src/lib/HST/Effect/InputFile.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | This module defines an effect for computations that can store the file
+--   path and the content of an input file.
+module HST.Effect.InputFile
+  ( -- * Effect
+    InputFile
+    -- * Actions
+  , setFile
+  , getFile
+    -- * Interpretations
+  , runInputFile
+  ) where
+
+import           Polysemy       ( Sem, makeSem, reinterpret )
+import           Polysemy.State ( State, evalState, get, put )
+
+-------------------------------------------------------------------------------
+-- Effect and Actions                                                        --
+-------------------------------------------------------------------------------
+-- | An effect capable of setting and getting the file path and the content of
+--   the input file.
+--
+--   While setting the input file, the content is converted to lines of code.
+data InputFile m a where
+  SetFile :: FilePath -> String -> InputFile m ()
+  GetFile :: InputFile m (FilePath, [String])
```

Comment:
I'm not sure whether a reader effect may not be more appropriate in this situation, i.e., remove the `SetFile` action. This would also be more flexible since we could add an alternative handler that reads the file using `IO` (plus caching).